lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -674,83 +843,446 @@ export default function getAPIImplementation(): APIInterface {
+                                return signUpResponse;
+                            }
+
+                            session = await Session.createNewSession(
+                                options.req,
+                                options.res,
+                                tenantId,
+                                signUpResponse.recipeUserId,
+                                {},
+                                {},
+                                userContext
+                            );
+
+                            return {
+                                status: "OK",
+                                session,
+                                user: signUpResponse.user,
+                            };
+                        } else {
+                            // active session
+                            let overwriteSessionDuringSignInUp = SessionRecipe.getInstanceOrThrowError().config
+                                .overwriteSessionDuringSignInUp;
+                            // MFA is disabled
+                            // Active session
+                            // Sign Up
+
+                            await checkIfSignUpIsAllowed(tenantId, email, userContext);
+
+                            let signUpResponse = await options.recipeImplementation.signUp({
+                                tenantId,
+                                email,
+                                password,
+                                shouldAttemptAccountLinkingIfAllowed: overwriteSessionDuringSignInUp,
+                                userContext,
+                            });
+                            if (signUpResponse.status === "EMAIL_ALREADY_EXISTS_ERROR") {
+                                return signUpResponse;
+                            }
+
+                            if (overwriteSessionDuringSignInUp) {
+                                session = await Session.createNewSession(
+                                    options.req,
+                                    options.res,
+                                    tenantId,
+                                    signUpResponse.recipeUserId,
+                                    {},
+                                    {},
+                                    userContext
+                                );
+                            }
+
+                            return {
+                                status: "OK",
+                                session,
+                                user: signUpResponse.user,
+                            };
+                        }
+                    } else {
+                        // MFA is active
+                        if (session === undefined) {
+                            // MFA is enabled
+                            // No Active session / first factor
+                            // Sign Up
+
+                            await checkIfSignUpIsAllowed(tenantId, email, userContext);
```

Comment:
we need to check for if first factor is allowed here before calling the signUp function

Reply:
Done